src/Controllers/LinkFieldController.php

```diff
@@ -253,6 +261,62 @@ public function save(array $data, Form $form): HTTPResponse
         return $response;
     }
 
+    /**
+     * Update Link Sort fields based frontend drag and drop sort logic
+     * /admin/linkfield/sort
+     */
+    public function linkSort()
+    {
+        $request = $this->getRequest();
+        // Check security token
+        if (!SecurityToken::inst()->checkRequest($request)) {
+            $this->jsonError(400, _t(__CLASS__ . '.INVALID_TOKEN', 'Invalid CSRF token'));
+        }
+        $json = json_decode($request->getBody() ?? '', true);
+        $newLinkIDs = (array) $json['newLinkIDs'];
+        if (empty($newLinkIDs)) {
+            $this->jsonError(400, _t('LinkField.BAD_DATA', 'Bad data'));
+        }
+        // Update index so that it can be used for the value of $sort
+        $newLinkIDs = array_values($newLinkIDs);
+        array_unshift($newLinkIDs, null);
+        unset($newLinkIDs[0]);
+        // Fetch and validate links
+        $links = Link::get()->filter(['ID' => $newLinkIDs])->toArray();
+        $linkIDToLink = [];
+        $ownerID = null;
+        $ownerRelation = null;
+        foreach ($links as $link) {
+            if (!$link->canEdit()) {
+                $this->jsonError(403, _t(__CLASS__ . '.UNAUTHORIZED', 'Unauthorized'));
+            }
+            if (is_null($ownerID)) {
+                $ownerID = $link->OwnerID;
+            } elseif ($link->OwnerID !== $ownerID) {
+                $this->jsonError(400, _t('LinkField.BAD_DATA', 'Bad data'));
+            }
+            if (is_null($ownerRelation)) {
+                $ownerRelation = $link->OwnerRelation;
+            } elseif ($link->OwnerRelation !== $ownerRelation) {
+                $this->jsonError(400, _t('LinkField.BAD_DATA', 'Bad data'));
+            }
```

Comment:
```suggestion
            if (is_null($ownerID)) {
                $ownerID = $link->OwnerID;
                $ownerRelation = $link->OwnerRelation;
            } elseif ($link->OwnerID !== $ownerID || $link->OwnerRelation !== $ownerRelation) {
                $this->jsonError(400, _t('LinkField.BAD_DATA', 'Bad data'));
            }
```
Simplifying this since it's really part of the same check.

Reply:
Updated